### PR TITLE
Adjust processing of all justifications

### PR DESCRIPTION
**Pull request by pacospace**
Signed-off-by: Francesco Murdaca <fmurdaca@redhat.com>

Related-To: https://github.com/thoth-station/report-processing/issues/87

### CODE REVIEW

File: producer.py [hunk 1]
```diff
@@ -72,28 +72,56 @@ async def main():
...
-                    summary_dataframe=adviser_summary_dataframe,
-                    date_filter=date,
-                    justification_type=justification_type,
-                    adviser_version=adviser_version,
-                )
+            for unique_message in adviser_dataframe["message"].unique():
+                subset_df = adviser_dataframe[
+                    (adviser_dataframe["message"] == unique_message)
+                    & (adviser_dataframe["date_"] == str(date.strftime("%Y-%m-%d")))
+                    & (adviser_dataframe["analyzer_version"] == adviser_version)
```
Comment (xtuchyna) [inline producer.py]: shouldn't be these logical `and` instead of bitwise `&`?

File: producer.py [hunk 2]
```diff
@@ -72,28 +72,56 @@ async def main():
...
+                ]
+
+                if not subset_df.empty:
+                    counts = subset_df.shape[0]
+
+                    types = [t for t in subset_df["type"].unique()]
+
+                    if len(types) > 1:
+                        _LOGGER.warning(f"type assigned to same message is different {types}")
```
Comment (xtuchyna) [inline producer.py]: I bluntly remember that using formatted strings actually slows down the `logger` in some way (making some unnecessary initializations i guess?), @fridex is that still true?
if so this should probably be
```suggestion
                        _LOGGER.warning("type assigned to same message is different %s", types)
```

File: producer.py [hunk 3]
```diff
@@ -72,28 +72,56 @@ async def main():
...
+        )
+
+    adviser_dataframe = Adviser._create_adviser_dataframe(justifications_collected)
+
+    if adviser_dataframe.empty:
+        _LOGGER.exception("No adviser documents identified")
+        return
+
+    adviser_dataframe["date_"] = [
+        pd.to_datetime(str(v_date)).strftime("%Y-%m-%d") for v_date in adviser_dataframe["date"].values
```
Comment (xtuchyna) [inline producer.py]: isn't this already done in https://github.com/thoth-station/report-processing/blob/7cad3661eb629f4ad5b66c5e47f1368a2cc5ac65/thoth/report_processing/components/adviser.py#L210 ?

wouldn't this be sufficient?
```suggestion
        v_date.strftime("%Y-%m-%d") for v_date in adviser_dataframe["date"].values
```

**Review by xtuchyna — COMMENTED** [review]
few small comments, otherwise lgtm

File: producer.py (hunk 1 above)
Comment (pacospace) [reply producer.py]: pandas requires `&` because the array in numpy are handled with that

File: producer.py (hunk 3 above)
Comment (pacospace) [reply producer.py]: Thanks!

File: producer.py (hunk 2 above)
Comment (pacospace) [reply producer.py]: Thanks!

File: producer.py (hunk 3 above)
Comment (pacospace) [reply producer.py]: Actually that is adjusted in new PR not yet merged, thanks!

**Comment by pacospace** [comment]
Ready, PTAL

**Pull request merged**